manifests/database.pp

```diff
@@ -1,17 +1,19 @@
 # Set up the PostgreSQL and Redis databases
 # @api private
 class pulpcore::database {
-  include postgresql::client
-  include postgresql::server
-  postgresql::server::db { $pulpcore::postgresql_db_name:
-    user     => $pulpcore::user,
-    password => postgresql_password($pulpcore::user, $pulpcore::postgresql_db_password),
-  }
+  if $pulpcore::postgresql_manage_db {
+    include postgresql::client
+    include postgresql::server
+    postgresql::server::db { $pulpcore::postgresql_db_name:
+      user     => $pulpcore::user,
+      password => postgresql_password($pulpcore::user, $pulpcore::postgresql_db_password),
+    }
 
-  pulpcore::admin { 'migrate --noinput':
-    unless      => 'python3-django-admin migrate --plan | grep "No planned migration operations"',
-    refreshonly => false,
-    require     => Postgresql::Server::Db[$pulpcore::postgresql_db_name],
+    pulpcore::admin { 'migrate --noinput':
```

Comment:
We'll still want to this to happen even if the database is external since it's handling migration of the data.

Reply:
@ehelms thanks. I thought so, but I ran into problems when I tried it.

I realized now, the issue was that `require     => Postgresql::Server::Db[$pulpcore::postgresql_db_name],` doesn't work when we aren't managing the DB. Flipping this to a `before` on the other resource instead seems to fix it.